Mathlib/Algebra/Module/Submodule/Lattice.lean

```diff
@@ -0,0 +1,425 @@
+    (↑(s.inf p) : Set M) = ⋂ i ∈ s, ↑(p i) := by
+  letI := Classical.decEq ι
+  refine' s.induction_on _ fun i s _ ih ↦ _
+  · simp
+  · rw [Finset.inf_insert, inf_coe, ih]
+    simp
+#align submodule.finset_inf_coe Submodule.finset_inf_coe
+
+@[simp]
+theorem infᵢ_coe {ι} (p : ι → Submodule R M) : (↑(⨅ i, p i) : Set M) = ⋂ i, ↑(p i) := by
+  rw [infᵢ, infₛ_coe] ; simp only [Set.mem_range, Set.interᵢ_exists, Set.interᵢ_interᵢ_eq']
+#align submodule.infi_coe Submodule.infᵢ_coe
+
+@[simp]
+theorem mem_infₛ {S : Set (Submodule R M)} {x : M} : x ∈ infₛ S ↔ ∀ p ∈ S, x ∈ p :=
+  Set.mem_interᵢ₂
+#align submodule.mem_Inf Submodule.mem_infₛ
+
+@[simp]
+theorem mem_infᵢ {ι} (p : ι → Submodule R M) {x} : (x ∈ ⨅ i, p i) ↔ ∀ i, x ∈ p i := by
+  rw [← SetLike.mem_coe, infᵢ_coe, Set.mem_interᵢ] ; rfl
+#align submodule.mem_infi Submodule.mem_infᵢ
+
+@[simp]
+theorem mem_finset_inf {ι} {s : Finset ι} {p : ι → Submodule R M} {x : M} :
+    x ∈ s.inf p ↔ ∀ i ∈ s, x ∈ p i := by
+  simp only [← SetLike.mem_coe, finset_inf_coe, Set.mem_interᵢ]
+#align submodule.mem_finset_inf Submodule.mem_finset_inf
+
+theorem mem_sup_left {S T : Submodule R M} : ∀ {x : M}, x ∈ S → x ∈ S ⊔ T := by
+  have : S ≤ S ⊔ T := le_sup_left
+  rw [LE.le] at this
+  exact this
+#align submodule.mem_sup_left Submodule.mem_sup_left
+
+theorem mem_sup_right {S T : Submodule R M} : ∀ {x : M}, x ∈ T → x ∈ S ⊔ T := by
+  have : T ≤ S ⊔ T := le_sup_right
+  rw [LE.le] at this
+  exact this
+#align submodule.mem_sup_right Submodule.mem_sup_right
+
+theorem add_mem_sup {S T : Submodule R M} {s t : M} (hs : s ∈ S) (ht : t ∈ T) : s + t ∈ S ⊔ T :=
+  add_mem (mem_sup_left hs) (mem_sup_right ht)
+#align submodule.add_mem_sup Submodule.add_mem_sup
+
+theorem sub_mem_sup {R' M' : Type _} [Ring R'] [AddCommGroup M'] [Module R' M']
+    {S T : Submodule R' M'} {s t : M'} (hs : s ∈ S) (ht : t ∈ T) : s - t ∈ S ⊔ T := by
+  rw [sub_eq_add_neg]
+  exact add_mem_sup hs (neg_mem ht)
+#align submodule.sub_mem_sup Submodule.sub_mem_sup
+
+theorem mem_supᵢ_of_mem {ι : Sort _} {b : M} {p : ι → Submodule R M} (i : ι) (h : b ∈ p i) :
+    b ∈ ⨆ i, p i :=
+  (le_supᵢ p i) h
+#align submodule.mem_supr_of_mem Submodule.mem_supᵢ_of_mem
+
+-- Porting note: commented out
+-- open BigOperators
+
+theorem sum_mem_supᵢ {ι : Type _} [Fintype ι] {f : ι → M} {p : ι → Submodule R M}
+    (h : ∀ i, f i ∈ p i) : (∑ i, f i) ∈ ⨆ i, p i :=
+  sum_mem fun i _ ↦ mem_supᵢ_of_mem i (h i)
+#align submodule.sum_mem_supr Submodule.sum_mem_supᵢ
+
+theorem sum_mem_bsupr {ι : Type _} {s : Finset ι} {f : ι → M} {p : ι → Submodule R M}
+    (h : ∀ i ∈ s, f i ∈ p i) : (∑ i in s, f i) ∈ ⨆ i ∈ s, p i :=
+  sum_mem fun i hi ↦ mem_supᵢ_of_mem i <| mem_supᵢ_of_mem hi (h i hi)
+#align submodule.sum_mem_bsupr Submodule.sum_mem_bsupr
+
+-- Porting note: Outdated comment? Lean 3 doesn't have this lemma in this file.
+/-! Note that `Submodule.mem_supr` is provided in `LinearAlgebra/Basic.lean`. -/
```

Comment:
```suggestion
/-! Note that `Submodule.mem_supᵢ` is provided in `LinearAlgebra/Basic.lean`. -/
```
Does this make the comment less outdated?

Reply:
Found it. Looks like `mem_supr` was refactored into a different file.
https://github.com/leanprover-community/mathlib/blob/fac369018417f980cec5fcdafc766a69f88d8cfe/src/linear_algebra/span.lean#L640

Reply:
Updated `Basic.lean` to `Span.lean`.